stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/resiliency/failover-connector.bal

```diff
@@ -0,0 +1,248 @@
+    action delete (string path, http:OutRequest request) (http:InResponse, http:HttpConnectorError) {
+        return performFailoverAction(path, request, null, DELETE_INT, failoverInferredConfig);
+    }
+
+    @Description {value:"The GET action implementation of the Failover Connector."}
+    @Param {value:"path: Resource path"}
+    @Param {value:"request: An OutRequest struct"}
+    @Return {value:"The InResponse struct"}
+    @Return {value:"Error occurred during the action invocation, if any"}
+    action get (string path, http:OutRequest request) (http:InResponse, http:HttpConnectorError) {
+        return performFailoverAction(path, request, null, GET_INT, failoverInferredConfig);
+    }
+}
+
+// populate boolean index array by looking at the Http status codes configured to be retry to get better performance
+// at runtime.
+function populateErrorCodeIndex (int[] errorCode) (boolean[] result) {
+    result = [];
+    foreach i in errorCode {
+        result[i] = true;
+    }
+    return result;
+}
+
+// performs execute action of the Failover connector. extract the corresponding http integer value representation
+// of the http verb and invokes the perform action method.
+function performExecuteAction (string path, http:OutRequest outRequest, http:InRequest inRequest,
+                               string httpVerb, FailoverInferredConfig failoverInferredConfig)
+(http:InResponse, http:HttpConnectorError) {
+    int httpVerbInt = extractHtpVerbIntValues(httpVerb);
+    return performFailoverAction(path, outRequest, inRequest, httpVerbInt, failoverInferredConfig);
+}
+
+// handles all the actions exposed through the Failover connector.
+function performFailoverAction (string path, http:OutRequest outRequest, http:InRequest inRequest,
+                                        int requestAction, FailoverInferredConfig failoverInferredConfig)
+                                                                        (http:InResponse, http:HttpConnectorError) {
+    boolean[] failoverCodeIndex = failoverInferredConfig.failoverCodesIndex;
+    int noOfEndpoints = lengthof (failoverInferredConfig.failoverClientsArray);
+    int currentIndex = 0;
+    int initialIndex = currentIndex;
+    int startIndex = -1;
+    int failoverInterval = failoverInferredConfig.failoverInterval;
+
+    FailoverConnectorError failoverConnectorError = {};
+    http:HttpClient[] failoverClients = failoverInferredConfig.failoverClientsArray;
+    http:HttpClient failoverClient = failoverClients[currentIndex];
+    http:InResponse inResponse;
+    http:HttpConnectorError httpConnectorError = {};
+    failoverConnectorError.httpConnectorError = [];
+
+    // When performing passthrough scenarios using Failover connector, message needs to be built before trying out the
+    // failover endpoints to keep the request message to failover the messages.
+    if (inRequest != null && FORWARD_INT == requestAction) {
+        blob binaryPayload = inRequest.getBinaryPayload();
```

Comment:
What is the purpose of this unused variable?

Reply:
When performing passthrough scenarios using the connector, the message needs to be built before trying out the failover to retain the incoming stream. If the message doesn't build prior to forwarding the request to the endpoint, the incoming stream will be consumed and it'll not be available for the failover.